cmd/geth/main.go

```diff
@@ -328,6 +331,29 @@ func startNode(ctx *cli.Context, stack *node.Node) {
 			}
 		}
 	}()
+
+	if exitWhenSynced := ctx.GlobalDuration(utils.ExitWhenSyncedFlag.Name); exitWhenSynced >= 0 {
+		go func() {
+			if ctx.GlobalString(utils.SyncModeFlag.Name) == "light" {
+				var lightEthereum *les.LightEthereum
+				if err := stack.Service(&lightEthereum); err != nil {
+					utils.Fatalf("LightEthereum service not running: %v", err)
+				}
+			} else {
+				var ethereum *eth.Ethereum
+				if err := stack.Service(&ethereum); err != nil {
+					utils.Fatalf("Ethereum service not running: %v", err)
+				}
+			}
+			var mux = stack.EventMux()
+			var sub = mux.Subscribe(downloader.DoneEvent{})
```

Comment:
The downloader done event is possibly not the best, because the downloader might be interrupted, or it might sync to an out of sync peer, report done, whilst missing blocks. My suggestion would be to subscribe to chain head events, and when the latest block's timestamp is recent enough, terminate.

Reply:
So to confirm, @rjl493456442   requested that I specifically use downloader.done.  I want to double check on this before I switch this

Reply:
I'm waiting for clarification, but in the meantime, I am looking into this.  As a part of that I would like to verify, by chain event do you mean the event related to the *TxPool object at pool.chainHeadCh? or a feed object?

Reply:
So my plan right now is to subscribe to ChainEvent within core the same way I am doing the download event and have something akin to
sub.Block.time() to get the the time and then compare that to the current time, if they are within a too be determined threshold I would then exit the synchronization.  This check would run every block as well I believe.  Would this be a proper way to proceed and have you had time to review the previous questions?

Reply:
I am having trouble finding a solution with a light footprint to implement it this way.  I've tried to access the downloader channel to keep up to dates on headers but I have had trouble accessing that.  I am right now looking into creating a new event within the downloader that will either fire off the header information with every block or will fire off an event when the header is caught up.  Let me know what your thoughts are.

Reply:
I pushed code to implement this, i have set a threshold for 10 minutes, you can view the explanation below

Reply:
After consulting with @rjl493456442 I am passing the latest header into the event and I am checking if it is within a threshold of 10 minutes of the current time I determine that it is the correct event and end it, if not nothing happens and it continues.  Let me know if this deals with the issue

Reply:
I am wrapping up testing on my end and will be resolving this out with a description in the main chat.  If passing the header into the downloader done event to check if we are within the threshold for a current block does not work please reopen this and let me know